Tools/autotest/sim_vehicle.py

```diff
@@ -851,6 +852,11 @@ def start_mavproxy(opts, stuff):
     location = find_location_by_name(find_autotest_dir(), cmd_opts.location)
     progress("Starting up at %s (%s)" % (location, cmd_opts.location))
 
+if cmd_opts.use_dir is not None:
+    new_dir = cmd_opts.use_dir
+    os.makedirs(new_dir)
```

Comment:
Assuming the exception can be caught distinctly as an it-already-exists exception, then you can just `pass` on that exception.  Further, using the exception avoids the obvious  TOCTTOU race.

I seriously think we should paint this bikeshed in nice, festive colours :-)